pkg/operator/configobserver/cloudconfig/sync_cloud_config.go

```diff
@@ -0,0 +1,121 @@
+package cloudprovider
+
+import (
+	"fmt"
+
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/util/sets"
+	corelisterv1 "k8s.io/client-go/listers/core/v1"
+
+	configv1 "github.com/openshift/api/config/v1"
+	configlistersv1 "github.com/openshift/client-go/config/listers/config/v1"
+	"github.com/openshift/library-go/pkg/operator/configobserver"
+	"github.com/openshift/library-go/pkg/operator/events"
+	"github.com/openshift/library-go/pkg/operator/resourcesynccontroller"
+)
+
+const (
+	configNamespace                 = "openshift-config"
+	machineSpecifiedConfigNamespace = "openshift-config-managed"
+	machineSpecifiedConfig          = "kube-cloud-config"
+)
+
+var validCloudProviders = sets.NewString("aws", "azure", "gce", "openstack", "vsphere")
```

Comment:
what does "valid" mean?

Reply:
seems like we could just call this `cloudProviders`, unless there might be some distinction in the future for "valid" and "experimental"(?) providers. wdyt @Danil-Grigorev ?